webapp/channels/src/components/admin_console/system_users/system_users_date_range_selector/index.tsx

```diff
@@ -0,0 +1,202 @@
+                            defaultMessage='All time'
+                        />
+                    }
+                    trailingElements={!inputValue && <i className='icon icon-check'/>}
+                    onClick={() => updateDateRange()}
+                />
+                <Menu.Item
+                    key={ReportDuration.Last30Days}
+                    id={ReportDuration.Last30Days}
+                    role='menuitem'
+                    labels={
+                        <>
+                            <FormattedMessage
+                                id='admin.system_users.date_range_selector.date_range.last_30_days'
+                                defaultMessage='Last 30 days'
+                            />
+                            <FormattedMessage
+                                id='admin.system_users.date_range_selector.date_range.sublabel'
+                                defaultMessage='{startDate} - {endDate}'
+                                values={{
+                                    startDate: formatDate(get30DaysBack(now)),
+                                    endDate: formatDate(now),
+                                }}
+                            />
+                        </>
+                    }
+                    trailingElements={inputValue === ReportDuration.Last30Days && <i className='icon icon-check'/>}
+                    onClick={() => updateDateRange(ReportDuration.Last30Days)}
+                />
+                <Menu.Item
+                    key={ReportDuration.PreviousMonth}
+                    id={ReportDuration.PreviousMonth}
+                    role='menuitem'
+                    labels={
+                        <>
+                            <FormattedMessage
+                                id='admin.system_users.date_range_selector.date_range.previous_month'
+                                defaultMessage='Previous month'
+                            />
+                            <FormattedMessage
+                                id='admin.system_users.date_range_selector.date_range.sublabel'
+                                defaultMessage='{startDate} - {endDate}'
+                                values={{
+                                    startDate: formatDate(getBeginningOfLastMonth(now)),
+                                    endDate: formatDate(getEndOfLastMonth(now)),
+                                }}
+                            />
+                        </>
+                    }
+                    trailingElements={inputValue === ReportDuration.PreviousMonth && <i className='icon icon-check'/>}
+                    onClick={() => updateDateRange(ReportDuration.PreviousMonth)}
+                />
+                <Menu.Item
+                    key={ReportDuration.Last6Months}
+                    id={ReportDuration.Last6Months}
+                    role='menuitem'
```

Comment:
do we need to explicitly have role? i thought it was added by default by the component

Reply:
I noticed other examples had them so not sure.

Reply:
https://github.com/mattermost/mattermost/blob/36983fddc63bcd1523a98d54f99cd4925b8eecf2/webapp/channels/src/components/menu/menu_item.tsx#L119

Its by default set